service/storage_service.cc

```diff
@@ -1676,12 +1698,15 @@ class topology_coordinator {
                 if (_topo_sm._topology.transition_nodes.empty()) {
                     builder.del_transition_state();
                 } else {
-                    builder.set_transition_state(topology::transition_state::write_both_read_old);
+                    builder.set_transition_state(topology::transition_state::tablet_draining);
```

Comment:
Why do we need to move into `tablet_draining` state during join?

Reply:
I wasn't sure if `publish_cdc_generation` is entered only during join by design. In general, it's safe to always go through tablet_drain, at worst it will do nothing. 

If you think it's safe, I will revert it.

Reply:
It is entered either during join or during cdc generation repair in which case it is a last state in the saga (that what 'if' checks). I prefer that each saga goes over only those state that are necessary.

Reply:
Ok